### PR TITLE
Remove hard dependency on FileIO.jl

### PR DESCRIPTION
**Pull request by kimikage**
This is in preparation for issue #65.
Unlike Colors.jl, I believe it is possible to remove the hard dependency on FilesIO.jl in a compatible manner, i.e. in FlameGraphs.jl v1.
This is because `FilesIO` must be loaded in order to use `File{format "JLPROF"}`.

For compatibility this depends on Requires.jl instead. However, it had already been in the dependencies indirectly via FileIO.jl.

### CODE REVIEW

**Comment by timholy** [comment]
Would you be OK with waiting for a release that contains your other fixes, and then we make a 1.1.0 release that makes Julia 1.10 (the future LTS) the minimum Julia version? That way we wouldn't need Requires.jl.

**Comment by kimikage** [comment]
My main aim is to decouple `Colors.jl` and I do not have a strong necessity for this PR itself.
So, I have no strong objection to bumping up the minimum version of Julia to v1.10.

However, I *personally* do not like the idea of changing the lower bounds of the compat entries within a single major version. Of course, this is a matter of "philosophy", not one that takes precedence over the practical issue of maintenance costs.

Since the usage of `Requires.jl` is idiomatic, *I* don't think the maintenance cost is too high.
In other words, it seems to me that the bumping could be done within `FlameGraphs.jl` v2.0.

One concern is that the decoupling of `Colors.jl` will take a bit of work, i.e., the v2.0 release might not be likely anytime soon.
